src/test/java/org/broadinstitute/hellbender/utils/config/ConfigUnitTest.java

```diff
@@ -0,0 +1,704 @@
+                },
+                // Three Paths:
+                {
+                        Arrays.asList( BasicTestConfig.class, GATKConfig.class, BasicTestConfigWithClassPathOverridesAndVariableFile.class ),
+                        Arrays.asList( GATKConfig.CONFIG_FILE_VARIABLE_FILE_NAME, GATKConfig.CONFIG_FILE_VARIABLE_CLASS_PATH, BasicTestConfigWithClassPathOverridesAndVariableFile.CONFIG_FILE_VARIABLE_FILE_NAME )
+                },
+        };
+    }
+
+    @Test(dataProvider= "createConfigListsAndResults")
+    public void testConfigUtilsGetConfigPathVariableNamesFromConfigClasses(final List<Class<?>> configurationClasses, final List<String> expectedFilePathVariables ) {
+        Assert.assertEquals( ConfigFactory.getInstance().getConfigPathVariableNamesFromConfigClasses(configurationClasses), expectedFilePathVariables );
+    }
+
+    @DataProvider
+    Object[][] createConfigFileAndVariableNames() {
+        return new Object[][] {
+                // Class with no @Sources annotation:
+                {
+                        BasicTestConfig.class,
+                        Collections.emptyList()
+                },
+                // Class with one @Sources annotation:
+                {
+                        BasicTestConfigWithClassPathOverridesAndVariableFile.class,
+                        Collections.singletonList( BasicTestConfigWithClassPathOverridesAndVariableFile.CONFIG_FILE_VARIABLE_FILE_NAME )
+                },
+                // Class with @Sources annotation with a file and a class path:
+                {
+                        GATKConfig.class,
+                        Arrays.asList( GATKConfig.CONFIG_FILE_VARIABLE_FILE_NAME, GATKConfig.CONFIG_FILE_VARIABLE_CLASS_PATH )
+                },
+        };
+    }
+
+    @Test(dataProvider = "createConfigFileAndVariableNames")
+    public void testGetSourcesAnnotationPathVariables(final Class<? extends Config> configClass, final List<String> expectedFilePathVariables ) {
+            Assert.assertEquals( ConfigFactory.getInstance().getSourcesAnnotationPathVariables(configClass), expectedFilePathVariables );
+    }
+
+    @DataProvider
+    Object[][] createConfigAndExpectedProperties() {
+
+        final LinkedHashMap<String, String> propertyMap1 = new LinkedHashMap<>();
+
+        propertyMap1.put("gatk_stacktrace_on_user_exception", "false");
+        propertyMap1.put("samjdk.use_async_io_read_samtools", "false");
+        propertyMap1.put("samjdk.use_async_io_write_samtools", "true");
+        propertyMap1.put("samjdk.use_async_io_write_tribble", "false");
+        propertyMap1.put("samjdk.compression_level", "1");
+        propertyMap1.put( "spark.kryoserializer.buffer.max", "512m" );
+        propertyMap1.put( "spark.driver.maxResultSize", "0" );
+        propertyMap1.put( "spark.driver.userClassPathFirst", "true" );
+        propertyMap1.put( "spark.io.compression.codec", "lzf" );
```

Comment:
Can you craft this test in such a way that it won't need to be updated every time we change the system properties in `GATKConfig`?

Reply:
Yeah - I can grab the properties from the configs using reflection and compare them to the created config instance properties.

It almost seems like it isn't testing anything if I'm checking the values in the class with the values in the class though.